File: number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.js
/**
 * @param {number[][]} rectangles
 * @return {number}
 */
var countGoodRectangles = function(rectangles) {
    
    let maxLength = [];
    
    for (let rect of rectangles){
        maxSide = Math.min.apply(null, rect)
        maxLength.push(maxSide)
    };
    
    let maxSideRectangle = Math.max.apply(null, maxLength);    
    
    return maxLength.filter(x => x==maxSideRectangle).length;
};

File: number-of-rectangles-that-can-form-the-largest-square/number-of-rectangles-that-can-form-the-largest-square.py
class Solution:
    def countGoodRectangles(self, rectangles: List[List[int]]) -> int:
        
        ## Create new list with maxLen sides for each rectangle      
        maxlen_list = [min(rect) for rect in rectangles]
        
        ## Find the greatest side 
        maxlen = max(maxlen_list)
        
        ## Return the count of rectangles with highest side
        return maxlen_list.count(maxlen)